caput/mpiarray.py

```diff
@@ -1069,6 +1153,203 @@ def _partition_io(self, skip=False, threshold=1.99):
         slices = [slice(start, end) for start, end in zip(starts, ends)]
         return split_axis, slices
 
+    @staticmethod
```

Comment:
This should probably just be a module level function. I think generally accepted style in Python is to avoid staticmethod's unless you have a good reason.